### ISSUE TITLE
Use Global Values when local values doesn't exist 

### ISSUE THREAD

**Issue by mcortinas** [helm/helm#5676]
I want use local values from alias dependency more priority than a common values. If the local value exist use this instead of global value.
Let me share an example of values.yaml in the parent chart
``` 
stable:
  key1: local_value1_stable_alias
global:
  key1: global_value1
```
The only option to do that I think is templating the subchart with this code
```
{{- if .Values.key1 }}
  key1: {{ .Values.key1 }}
{{- else }}
  key1: {{ .Values.global.key1 }}
{{- end }}
```
Basically, we want use a global values when local values is not set. 

I tried the workaround commented on https://github.com/helm/helm/issues/2076 using the global inside a local valuas from alias (called stable in this example)
```
stable:
  global:
    key4: value4_from_megachart_stable_global
global:
  key4: value4_from_megachart
```
This is not a perfect workaround due to the global variables has more priority than local variables, let me define a common section in the same values file but with more priority.

Do you know some  go functions in order to be subchart template code more simple???

**Comment by mcortinas** [comment]
I also tried use default from http://masterminds.github.io/sprig/defaults.html with this code
```
key3: {{ default .Values.key3 .Values.global.key3 }}
```
and this:
```
key3: {{ default .Values.global.key3 .Values.key3 }}
```
without lucky, It seems only print local value if global is not defined.

**Comment by mattfarina** [comment]
@mcortinas If I understand it right, you want the local variable to have precedence over the global one. Two things...

First, I would be very curious to know your use case so we can better understand the need. We're always curious to understand what people are doing and why they're doing it.

Second, you could create a function in the _helpers.tpl file to pick the local one if it has a value or fall back to a global one. This might provide a quick solution you can use now.

**Comment by jgleonard** [comment]
@mattfarina I was thinking about a similar functionality. 

I have a few standalone charts that can also be grouped together under a parent chart.  Two scenarios here...

1) In the case that they are being called as subcharts, I would want to use a global rather than duplicate values for each child, but I wouldn't want to be forced into using globals in the children when they're being deployed standalone.

2) Additionally, I could envision a scenario where a value passed to a subchart would be assumed to override the global at that same level, but since Helm doesn't appear to make that distinction, that level of precedence isn't possible.

```
# chartA
food: ham
vessel: plate
```

```
# chartB
food: eggs 
vessel: bowl
```

```
# parent
chartA:
  food: bacon

chartB:
  vessel: plate

global:
  food: sausage
```

Deployed with the parent, I could arguments for expecting a plate of bacon and a plate of sausage, but given the way that helm sets values, you'd end up with 2 plates of sausage.  Apologies for the poor example, but tried to keep it simple.

**Comment by jgleonard** [comment]
Also reminds me of https://github.com/helm/helm/issues/2492

**Comment by jfcoz** [comment]
This could also be usefull to pass the nodeSelector to all dependency, with helm --set global.nodeSelector=…

Each dependency would use it, unless dependency.nodeSelector is defined

**Comment by flutt13** [comment]
Don't know if the question is still valid, but I think I solved this problem by using merge function

values.yaml:
```
global:
  defaults:
    key1: value1
    key2: value2

subchart_name:
  defaultOverrides:
    key1: test
 ```

And then in any of the subchart templates
```
{{- $properties := merge .Values.defaultOverrides $.Values.global.defaults }}
{{ toYaml $properties }}
```

which should print

```
key1: test
key2: value2
```

This will make subchart value override global value (this is because we passed local .Values as a first argument to merge which takes higher precedence).
As a bonus no need to use the ugly if structure or even default function. The value set in global section acts as a default if it's not overridden in subchart values.
